app/models/invoice.rb

```diff
@@ -1,4 +1,21 @@
 class Invoice < ActiveRecord::Base
   has_many :line_items
   belongs_to :client
+
+  accepts_nested_attributes_for :line_items, :allow_destroy => true
+  acts_as_sequenced column: :invoice_number, start_at: 1000
+
+  def self.find_next_available_number_for(default=1000)
+    (self.maximum(:invoice_number) || default).succ
+  end
+
+  def date_of_issue=(val)
+    date = Date.strptime(val, "%m/%d/%Y") if val.present?
```

Comment:
Create method `parse_date` with following parameter `parse_date(date, format = '%m/%d/%Y')`  and reuse it withing this class.